SteamKit2/SteamKit2/Networking/Steam3/UdpConnection.cs

```diff
@@ -137,6 +137,18 @@ public void Connect(EndPoint endPoint, int timeout)
             inSeqAcked = 0;
             inSeqHandled = 0;
 
+            if (endPoint is DnsEndPoint dnsEndPoint)
+            {
+                var resolved = Dns.GetHostAddresses(dnsEndPoint.Host, AddressFamily.InterNetwork);
+
+                if (resolved == null || resolved.Length == 0)
+                {
+                    throw new ArgumentException("Failed to resolve given dns endpoint.", nameof(endPoint));
+                }
+
+                endPoint = new IPEndPoint(resolved[0], dnsEndPoint.Port);

```

Comment:
Should we select random if it resolves to multiple hosts?

Reply:
I'm not really expecting UDP servers to live for long, so I don't care too much 🤣

Reply:
lol valid